src/client/Microsoft.Identity.Client/Platforms/Android/AndroidBrokerHelper.cs

```diff
@@ -159,7 +166,14 @@ public async Task<string> GetBrokerAuthTokenSilentlyAsync(IDictionary<string, st
 
         public void GetBrokerAccountInfo(IDictionary<string, string> brokerPayload, Activity callerActivity)
         {
-            var loginHint = GetValueFromBrokerPayload(brokerPayload, BrokerParameter.LoginHint);
+            string loginHint = string.Empty;
+            if (string.IsNullOrEmpty(GetValueFromBrokerPayload(brokerPayload, BrokerParameter.LoginHint)))
+            {
+                _logger.Error(MsalErrorMessage.NoUPNOrAccountIDForSilentAuth);
+                throw new MsalClientException(MsalError.NoUsernameOrAccountIDProvidedForSilentAndroidBrokerAuthentication, MsalErrorMessage.NoUPNOrAccountIDForSilentAuth);
```

Comment:
>MsalClientException [](start = 26, length = 19)

I guess i'm still confused here. so if there is no login hint, we throw an exception, but how does the developer resolve the issue? if they are doing 
try { ATSilent() } catch(MsalUiRequiredException ){ ATInteractive() }

They won't catch this error, unless later we are bubbling it up as an MsalUiRequiredException. Or is it required to provide a login hint w/the silent flow? if so, we should check earlier and fail if it's not provided.

Reply:
also, this would be a breaking change, right?

---
In reply to: [382131373](https://github.com/AzureAD/microsoft-authentication-library-for-dotnet/pull/1649#discussion_r382131373) [](ancestors = 382131373)

Reply:
Discussed offline. removing exception